src/Avalonia.Visuals/Media/TextFormatting/Unicode/BinaryReaderExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// RichTextKit
+// Copyright © 2019 Topten Software. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License"); you may 
+// not use this product except in compliance with the License. You may obtain 
+// a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the 
+// License for the specific language governing permissions and limitations 
+// under the License.
+// Copied from: https://github.com/toptensoftware/RichTextKit
+
+using System;
+using System.IO;
+
+namespace Avalonia.Media.TextFormatting.Unicode
+{
+    internal static class BinaryReaderExtensions
+    {
+        public static int ReadInt32BE(this BinaryReader reader)
+        {
+            var bytes = reader.ReadBytes(4);
+            Array.Reverse(bytes);
```

Comment:
I know this is not your code, but this code won't work correctly on an actual big endian machine (even though it may never run on such). It should only reverse when on little endian machine.